docs: Change license of Mangrove core to Business Source License 1.1

Louis-Amas opened the pull request.

Louis-Amas committed:
fix: update-license for mangroce core files

Louis-Amas committed:
feat: improve explanation of LICENSE

espendk reviewed (commented):
It seems some of the files that should be covered by BUSL are missing:

![billede](https://user-images.githubusercontent.com/3458856/236172106-a869603f-7e8a-4575-8ba9-ebd48e8bae9c.png)

espendk reviewed (changes requested):
The LICENSE file should still contain all the other licenses (which we still refer to in other files), the BUSL license should just be added (instead of replacing everything).

Louis-Amas commented:
> It seems some of the files that should be covered by BUSL are missing:
> 
> ![billede](https://user-images.githubusercontent.com/3458856/236172106-a869603f-7e8a-4575-8ba9-ebd48e8bae9c.png)

I used the one you updated below in the Asana task
![billede](https://user-images.githubusercontent.com/28564552/236172897-e4bfd688-6c67-48e5-9114-0b0bc39e2ada.png)

espendk commented:
> > It seems some of the files that should be covered by BUSL are missing:
> > ![billede](https://user-images.githubusercontent.com/3458856/236172106-a869603f-7e8a-4575-8ba9-ebd48e8bae9c.png)
> 
> I used the one you updated below in the Asana task ![billede](https://user-images.githubusercontent.com/28564552/236172897-e4bfd688-6c67-48e5-9114-0b0bc39e2ada.png)

which lists the exact same files as your list ;-)

Louis-Amas committed:
fix: AbstractMangrove update LICENSE

Louis-Amas committed:
fix: periphery update license

espendk commented on src/AbstractMangrove.sol:
```diff
@@ -1,4 +1,4 @@
-// SPDX-License-Identifier:	AGPL-3.0
+// SPDX-License-Identifier:	BUSL-1.1
```
```suggestion
// SPDX-License-Identifier: BUSL-1.1
```

espendk commented on LICENSE:
I'll push a revision of this.

espendk committed:
chore: Update LICENSE

espendk committed:
chore: Update LICENSE with input from Phil

espendk committed:
chore: Change tab to whitespace

espendk committed:
chore: Update LICENSE with input from Phil

espendk committed:
chore: Update LICENSE with input from Phil

espendk committed:
chore: Update LICENSE with input from Phil

espendk reviewed (approved):
Phil has reviewed and approves.

espendk committed:
Merge branch 'develop' into feat/update-license

espendk committed:
chore: Update license field in package.json

As the license differs for different parts of the package, an SPDX expression is insufficient to specify the license of the package. Instead, we refer to the LICENSE file as instructed in the documentation for package.json: https://docs.npmjs.com/cli/v9/configuring-npm/package-json#license
